File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> indexes;
        for(int i = 0; i < nums.size(); ++i) {
            int current = nums[i];
            int complement = target - current;
            if(indexes.count(complement)) return {indexes[complement], i};
            indexes[current] = i;
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = dict()
        for i in range(len(nums)):
            current = nums[i]
            complement = target - current
            if complement in d: return [i, d[complement]]
            d[current] = i
        return []
        